bundles/org.openhab.binding.haassohnpelletstove/src/main/java/org/openhab/binding/haassohnpelletstove/internal/HaasSohnpelletstoveHandler.java

```diff
@@ -0,0 +1,327 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.haassohnpelletstove.internal;
+
+import static org.openhab.binding.haassohnpelletstove.internal.HaasSohnpelletstoveBindingConstants.*;
+
+import java.text.DecimalFormat;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.measure.Unit;
+import javax.measure.quantity.Temperature;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HaasSohnpelletstoveHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Christian Feininger - Initial contribution
+ */
+@NonNullByDefault
+public class HaasSohnpelletstoveHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(HaasSohnpelletstoveHandler.class);
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    private HaasSohnpelletstoveConfiguration config = new HaasSohnpelletstoveConfiguration();
+    boolean resultOk = false;
+
+    private HaasSohnpelletstoveJSONCommunication serviceCommunication;
+
+    private boolean automaticRefreshing = false;
+
+    private Map<String, Boolean> linkedChannels = new HashMap<String, Boolean>();
+
+    public HaasSohnpelletstoveHandler(Thing thing) {
+        super(thing);
+        serviceCommunication = new HaasSohnpelletstoveJSONCommunication();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(CHANNELPOWER)) {
+            String postData = null;
+            if (command.equals(OnOffType.ON)) {
+                postData = "{\"prg\":true}";
+            } else if (command.equals(OnOffType.OFF)) {
+                postData = "{\"prg\":false}";
+            }
+            if (postData != null) {
+                logger.debug("Executing {} command", CHANNELPOWER);
+                updateOvenData(postData);
+            }
+        } else if (channelUID.getId().equals(CHANNELSPTEMP)) {
+            if (command instanceof QuantityType<?>) {
+                @SuppressWarnings("unchecked")
+                QuantityType<Temperature> value = (QuantityType<Temperature>) command;
```

Comment:
You can avoid the supression by using the wildcard.
```suggestion
                QuantityType<?> value = (QuantityType<?>) command;
```